Game/Enemies/GenericEnemy.cs

```diff
@@ -1,11 +1,14 @@
 ﻿using Microsoft.Xna.Framework;
 using MainGame.SpriteHandlers;
+using MainGame.Audio;
 
 namespace MainGame.Enemies
 {
 	public abstract class GenericEnemy : IEnemy
 	{
-        public static readonly int ImmunityFrame = 10;
+        public AudioManager AudioManager { get; set; }
```

Comment:
Instead of having each concrete class that inherits from GenericEnemy initialize AudioManager by itself, it might be better to have it initialized here, in the abstract class. It helps ensure that when it's used down below, it isn't a null reference. You can make AudioManager a param in the initialization function of this abstract class and then every concrete class that inherits from it have to pass in a value (via the "base" keyword) .